packages/core-mobile/ios/NotificationServiceExtension/NotificationService.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  NotificationService.swift
+//  NotificationServiceExtension
+//
+//  Created by Ruijia Lin on 9/11/24.
+//
+
+import UserNotifications
+
+class NotificationService: UNNotificationServiceExtension {
+
+    var contentHandler: ((UNNotificationContent) -> Void)?
+    var bestAttemptContent: UNMutableNotificationContent?
+    let defaults = UserDefaults(suiteName: "group.org.avalabs.corewallet")
+
+    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        self.contentHandler = contentHandler
+        bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
+      var count: Int = defaults?.value(forKey: "badgeCount") as? Int ?? 0
```

Comment:
we can move this variable inside the `if let bestAttemptContent = bestAttemptContent` statement.

Reply:
[updated](238952ac669a47143fc5953850670c165d139c21)